sunpy/net/base_client.py

```diff
@@ -31,9 +109,56 @@ def __init_subclass__(cls, *args, **kwargs):
         This means that Fido can use the clients internally.
         """
         super().__init_subclass__(**kwargs)
+
         # We do not want to register GenericClient since its a dummy client.
-        if cls.__name__ != 'GenericClient':
-            cls._registry[cls] = cls._can_handle_query
+        if cls.__name__ in ('GenericClient'):
+            return
+
+        cls._registry[cls] = cls._can_handle_query
+
+        if hasattr(cls, "_attrs_module"):
+            from sunpy.net import attrs
+
+            name, module = cls._attrs_module()
+            module_obj = importlib.import_module(module)
+
+            existing_mod = getattr(attrs, name, None)
+            if existing_mod and existing_mod is not module_obj:
+                raise NameError(f"{name} has already been registered as an attrs name.")
+
+            setattr(attrs, name, module_obj)
+
+            if name not in attrs.__all__:
+                attrs.__all__.append(name)
+
+        # Register client attrs after it has regsitered its own attrs
+        from sunpy.net import attr
+        attr.Attr.update_values(cls.register_values())
+
+    def __repr__(self):
+        """
+        Returns the normal repr plus the pretty client __str__.
+        """
+        return object.__repr__(self) + "\n" + str(self)
+
+    def __str__(self):
```

Comment:
We should check that we aren't overloading this in any of the clients (i.e. I don't think it's working for VSO.)

Reply:
I don't see VSO with its own ``__str__`` method.

Reply:
I can print the VSOClient and get an output here. 

```python
In [5]: vso.VSOClient()                                                                                                                                                                                                                                        
Out[5]: 
<sunpy.net.vso.vso.VSOClient object at 0x7fe322173af0>
VSOClient

VSO Client

Attr Type                   Name                                                                                                       Description                                                                                       
---------- ------------------------------------- ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Detector                                   BFI                                                                                                                                                                                     None
  Detector                                  CTOF                                                                                                                                                  Charge Time Of Flight mass spectrometer
  Detector                                   HI1                                                                                                                                                                    Heliospheric Imager 1
```
etc